devto-draft.md

```diff
@@ -1 +1,393 @@
+
+In React, this often meant making another component called "group(s)" and maybe one called "group".
+
+Vue, you can just dump the `v-for`, `v-if` etc right inside your HTML. Your logic coherently just makes sense looking at it.
+
+Alternatively, you can write your own directives! This lets you attach functionality to page if necessary, making it easier to develop faster
+
+One great use case for startups is to develop a reporting system. If a user clicks on a feature that doesn't yet exist, you can add a popup modal saying "Sorry we're working hard on this new feature! Come back later". In the background, you can make Axios request to notify users really do want that feature.
+
+## Mixins
+
+Some companies swear by these. It's a reusable code snippet library that can be implemented across multiple components. React out of the box does not have such a feature, you can alternatively use callback functions instead that are made globally available
+
+## Emitting Events
+
+Props are passed from parent to child component, via one-way data binding. To handle business logic at the parent level, from the child component, you generally emit events.
+
+With Vue, there are two different ways.
+
+1. You can pass a reference to the function from parent to child
+2. Emit an event from the child and capture the response in parent
+
+Most applications generally use #2. Example from a button that triggers a method in the parent component
+
+```vue
+<!-- Parent Component -->
+<template>
+  <ChildComponent @event-emitted="_handleUpdatedEvent" />
+</template>
+
+<script>
+import ChildComponent from "./components/ChildComponent.vue";
+
+export default {
+  components: {
+    ChildComponent
+  },
+  methods: {
+    _handleUpdatedEvent(val) {
+      console.log("Button was clicked");
+    }
+  }
+};
+</script>
+```
+
+```vue
+<!-- Child Component -->
+<template>
+  <div class="hello">
+    <button @click="$emit('event-emitted')">Click Me</button>
+  </div>
+</template>
+```
+
+## Global State Management
+
+Vue's CLI comes with the ability to add Vuex out of the box.
+
+In React you pass a new set of state using `setState()`, Redux only extends this through global dispatches/flux architecture.
+
+In Vue, you mutate objects behind the scene. Vuex is no different but at the global level.
+
+Vue natively ships with the ability to create a global event bus, which is similar to React's context API.
+
+## Final Thoughts
+
+Learning Vue has been a great way to learn new programming paradigms.
+
+Many principles in React translate over to Vue. There are a few things that differ, namely:
+
+- You don't replace your state, you mutate it. There's data reactivity / async gotchas from these
+- New functionality. There are computed properties, watchers, mixins, etc that don't exist in React out of the box
+- You write HTML and styles the way you would traditionally in a regular index.html page
+- PropTypes are enforced in Vue, it's optional in React
```

Comment:
Are they called PropTypes in Vue as well?

Reply:
Yeah it's called Prop Types as well